### ISSUE TITLE
Discussion: Should we facilitate add-ins?

### ISSUE THREAD

**Issue by mansellan** [rubberduck-vba/Rubberduck#4678]
Meant as an open discussion, rather than a quick yes/no. This is a proposition, feel free to challenge or disagree.

Rubberduck's guiding mission is to bring the VBE into the 21st century. To facilitate this, we might allow others to bring ideas which we may not necessarily endorse, prioritize, or have capacity to implement and support. For this reason, ReSharper supports add-ins, and we could too.

This would be more than just tolerating other COM add-ins. In particular, we could provide a standard API to:
- Acquire windows (dialogs, toolwindows, panes) tailored to supporting WPF controls
- Insert items into the RD menu (without having to navigate COM CommandBars)
- Get a node in the RD Settings dialog
- Receive notifications (or Observables?) for VBE and/or RD events

In short, bringing VBE **add-ins** into the 21st century.

So the question is, should we?

**Comment by bclothier** [comment]
I could have sworn there was an issue already but I am unable to find that issue where such idea was discussed (it might have been about plugins or may be specific to inspections only).

Anyway, I think a major prerequisite that must be met before we can even start to talk of addin/plugin anything is that we have a mean of loading those in an isolated manner. If we can't provide a stable environment that doesn't go tumbling down just because someone sniffled, then nobody is going to want to use the Rubberduck core product. That implies having the COM shim project ( #4041 ), so that we can load the addin in a separate domain and thus punt them gracefully without going down in the fire with them. 

I think that point needs to be satisfied first before we can start looking at offering other people Rubberduck as a platform, rather than as a software, and that's not even considering the other issues associating with offering RD as a platform such as the accountability for the maintenance of code quality.

**Comment by mansellan** [comment]
Just on that last point - I think a dialog with author terms predicated on conditional user acceptance (or not) for add-in installation should be enough to direct accountability to the author and their license.

AFAICT - the pop-ups NuGet spews.
